docs/en/02_Developer_Guides/14_Files/04_File_Storage.md

```diff
@@ -53,32 +53,100 @@ storage backend to determine how variants are managed.
+They are stored as you'd expect on the filesystem: In their folder, by their file name.
 
-For instance, this is a typical disk content:
+```
+assets/
+    my-public-folder/
+        my-public-file.jpg
+```
+
+The URL for this file will match the physical location on disk:
+`http://www.example.com/assets/my-public-folder/my-public-file.jpg`.
+
+## Variant file paths (e.g. resized images)
+
+Each file can have variants, most commonly resized versions of an image.
+These can be generated by resizing an image in the CMS rich text editor,
+through template logic, or programmatically with PHP.
+They are stored in the same folder alongside the original file,
+but contain a special variant suffix.
+
+```
+assets/
+    my-public-folder/
+        my-public-file.jpg
+        my-public-file__FitWzYwLDYwXQ.jpg
+```
+
+The URL for this file will match the physical location on disk:
+`http://www.example.com/assets/my-public-folder/my-public-file__FitWzYwLDYwXQ.jpg`.
+
+Note: Before 4.4.0, public files were stored with a content hash by default
+(see [Protected file paths](#protected-file-paths)).
+
+## Protected file paths {#protected-file-paths}
+
+Uploaded files are protected by default, which puts them in a "draft" mode
+that requires permissions to view them. Protected files can also be published
+but access restricted. In either case, they're stored in a special `assets/.protected` folder.
+In this case, they're stored in a folder matching the truncated hash of the file's content.
 
 ```
 assets/
-    Uploads/
-        b63923d8d4/
-            BannerHeader.jpg
-            BannerHeader__FitWzYwLDYwXQ.jpg
+    my-public-folder/
+        my-public-file.jpg
+        my-public-file__FitWzYwLDYwXQ.jpg
+    .protected/
+        my-protected-folder/
+            b63923d8d4/
+                my-protected-file.jpg
+                my-protected-file__FitWzYwLDYwXQ.jpg
 ```
 
 This corresponds to a file with the following properties:
 
-- Filename: Uploads/BannerHeader.jpg
+- Filename: my-protected-folder/my-protected-file-hash/my-protected-file.jpg
 - Hash: b63923d8d4089c9da16fbcbcdfef3e1b24806334 (trimmed to first 10 chars)
-- Variant: FitWzYwLDYwXQ (corresponds to Fit[60,60])
+- Variant: FitWzYwLDYwXQ (corresponds to `Fit[60,60]`)
 
-The URL for this file will match the physical location on disk:
-`http://www.example.com/assets/Uploads/b63923d8d4/BannerHeader__FitWzYwLDYwXQ.jpg`.
+The URL for this file will not match the physical location on disk.
+It leaves out the `.protected/` folder, and leaves that to SilverStripe's integrated routing:
+`http://www.example.com/assets/my-protected-folder/b63923d8d4/my-protected-file.jpg`.
 
 For more information on how protected files are stored see the [file security](/developer_guides/files/file_security)
 section.
 
+## Versioned file paths
+
+Older versions of file contents are kept in the `.protected` folder,
+following the same rules as [protected file paths](#protected-file-paths).
```

Comment:
That kind of make it sounds like this is the default behaviour, which it isn't. It's only going to do that `keep_archived_assets` is enabled.

Reply:
Good catch, fixed